pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -86,11 +90,17 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 				},
 			},
 		}
+
 		if !f.env.IsLocalDevelopmentMode() /* not local dev or CI */ {
 			doc.OpenShiftCluster.Properties.FeatureProfile.GatewayEnabled = true
 		}
 	}
 
+	err = validateIdentityUrl(doc.OpenShiftCluster, identityURL, isCreate)
```

Comment:
Nit: in general, it's not good to have flag arguments: https://martinfowler.com/bliki/FlagArgument.html

So maybe to follow the paradigm in this function, we can change validation based on `isCreate`:

```
if isCreate {
...
} else {
...
}
```

Reply:
Do you think it would be better to split validateIdentityUrl into two functions? I ask because either way, we will need the `isCreate` logic in place as you pointed out, it's just a matter of where it goes.

Reply:
I'd like us to clean this up before GA (or soon after) but not hold up overall progress. Since this a current norm in our codebase we can acknowledge that it should be improved and move on to mission critical pieces. A good fist issue for a new joiner would be this refactor.